### DISCUSSION TITLE
`pnpm add` slow-down in v8 vs v7

### DISCUSSION THREAD

**Discussion by vjpr** [pnpm/pnpm#6923] [General]
`pnpm add` for a single package used to take <10s in v7. Now it's over a minute which makes it hard to work.

I used the `--reporter=ndjson` to time how long each stage takes:

```
scope
scope 2.3s
pnpm
pnpm 194ms
package-manifest
package-manifest 447ms
context
context 25ms
stage resolution_started
stage resolution_started 22.9s
install-check
install-check 22.9s
package-manifest
package-manifest 2.4s
stage resolution_done
stage resolution_done 176ms
stats
stats 1ms
stage importing_started
stage importing_started 186ms
stats
stats 28ms
link
link 1ms
stage importing_done
stage importing_done 28ms
pnpm
pnpm 136ms
link
link 417ms
global
global 169ms
bins-conflict
bins-conflict 68ms
global
global 562ms
lockfile
lockfile 606ms
bins-conflict
bins-conflict 87ms
global
global 472ms
bins-conflict
bins-conflict 1ms
global
global 276ms
bins-conflict
bins-conflict 1ms
global
global 76ms
summary
summary 22ms
peer-dependency-issues
peer-dependency-issues 1ms
execution-time
```

Looks like these stages are the main culprits...although my way of timing may not be totally accurate. I'm just measuring the time between each message and a different type of message.

```
stage resolution_started
stage resolution_started 22.9s
install-check
install-check 22.9s
```

I already tried setting: `verify-store-integrity=false`

Any guidance?

Is there a better way to time each stage?

I couldn't find any usage of `node:diagnostics_channel` or `performance.now`...

Related: https://github.com/orgs/pnpm/discussions/3343

**Comment by YFengFly** [comment]
Hi~
Maybe you can try this method:
https://github.com/orgs/pnpm/discussions/8826

**Reply by zkochan** [reply]
I don't think that is related to this issue. This issue is probably about the added complexity of proper peer dependencies resolution and deduplication. But we might have optimized it since then.
